src/networking.c

```diff
@@ -4667,16 +4787,18 @@ int processIOThreadsReadDone(void) {
             c->flag.pending_command = 1;
         }
 
-        size_t list_length_before_command_execute = listLength(server.clients_pending_io_read);
-        if (processPendingCommandAndInputBuffer(c) == C_OK) {
-            beforeNextClient(c);
-        }
-        if (list_length_before_command_execute != listLength(server.clients_pending_io_read)) {
-            /* A client was unlink from the list possibly making the next node invalid */
+        addCommandToBatchAndProcessIfFull(c);
+
+        /* There is a possibility that a client was closed during the latest batch processing.
+         * In this case the next node may be invalid */
+        if (batch.client_closed_during_batch_execution) {
             next = listFirst(server.clients_pending_io_read);
+            batch.client_closed_during_batch_execution = 0;
```

Comment:
To simplify this code, should we just always call listFirst(server.clients_pending_io_read) instead of listNext?